warp/handlers/signature_request_test.go

```diff
@@ -0,0 +1,202 @@
+			require.NoError(t, err)
+
+			test.verifyStats(t, handler.stats)
+
+			// If the expected response is empty, assert that the handler returns an empty response and return early.
+			if len(expectedResponse) == 0 {
+				require.Len(t, responseBytes, 0, "expected response to be empty")
+				return
+			}
+			var response message.SignatureResponse
+			_, err = message.Codec.Unmarshal(responseBytes, &response)
+			require.NoError(t, err, "error unmarshalling SignatureResponse")
+
+			require.Equal(t, expectedResponse, response.Signature[:])
+		})
+	}
+}
+
+func TestBlockSignatureHandler(t *testing.T) {
+	database := memdb.New()
+	snowCtx := snow.DefaultContextTest()
+	blsSecretKey, err := bls.NewSecretKey()
+	require.NoError(t, err)
+
+	warpSigner := avalancheWarp.NewSigner(blsSecretKey, snowCtx.NetworkID, snowCtx.ChainID)
+	blkID := ids.GenerateTestID()
+	testVM := &block.TestVM{
+		TestVM: common.TestVM{T: t},
+		GetBlockF: func(ctx context.Context, i ids.ID) (snowman.Block, error) {
+			if i == blkID {
+				return &snowman.TestBlock{
+					TestDecidable: choices.TestDecidable{
+						IDV:     blkID,
+						StatusV: choices.Accepted,
+					},
+				}, nil
+			}
+			return nil, errors.New("invalid blockID")
+		},
+	}
+	backend := warp.NewBackend(
+		snowCtx.NetworkID,
+		snowCtx.ChainID,
+		warpSigner,
+		testVM,
+		database,
+		100,
+	)
+
+	signature, err := backend.GetBlockSignature(blkID)
+	require.NoError(t, err)
+	unknownMessageID := ids.GenerateTestID()
+
+	emptySignature := [bls.SignatureLen]byte{}
+
+	tests := map[string]struct {
+		setup       func() (request message.BlockSignatureRequest, expectedResponse []byte)
+		verifyStats func(t *testing.T, stats *handlerStats)
+	}{
+		"known block": {
+			setup: func() (request message.BlockSignatureRequest, expectedResponse []byte) {
+				return message.BlockSignatureRequest{
+					BlockID: blkID,
+				}, signature[:]
+			},
+			verifyStats: func(t *testing.T, stats *handlerStats) {
+				require.EqualValues(t, 0, stats.messageSignatureRequest.Count())
+				require.EqualValues(t, 0, stats.messageSignatureHit.Count())
+				require.EqualValues(t, 0, stats.messageSignatureMiss.Count())
+				require.EqualValues(t, 1, stats.blockSignatureRequest.Count())
+				require.EqualValues(t, 1, stats.blockSignatureHit.Count())
+				require.EqualValues(t, 0, stats.blockSignatureMiss.Count())
+			},
+		},
+		"unknown block": {
+			setup: func() (request message.BlockSignatureRequest, expectedResponse []byte) {
+				return message.BlockSignatureRequest{
+					BlockID: unknownMessageID,
+				}, emptySignature[:]
+			},
+			verifyStats: func(t *testing.T, stats *handlerStats) {
+				require.EqualValues(t, 0, stats.messageSignatureRequest.Count())
+				require.EqualValues(t, 0, stats.messageSignatureHit.Count())
+				require.EqualValues(t, 0, stats.messageSignatureMiss.Count())
+				require.EqualValues(t, 1, stats.blockSignatureRequest.Count())
```

Comment:
should we add `require.EqualValues(t, stats.messageSignatureRequestDuration.Value(), int64(0))`  and `require.Greater(t, stats.blockSignatureRequestDuration.Value(), int64(0))` ? 
https://github.com/ava-labs/subnet-evm/blob/32514d6c7c6523d08a07eadda9919ef2253232de/warp/handlers/signature_request_test.go#L161

Reply:
This was explicitly removed because it fails on windows (it doesn't register the amount of elapsed time as non-zero). Made the same change on Subnet-EVM to keep them in sync.

Good 👁️